shap/explainers/_deep/deep_tf.py

```diff
@@ -130,33 +130,35 @@ def __init__(self, model, data, session=None, learning_phase_flags=None):
         # if no learning phase flags were given we go looking for them
         # ...this will catch the one that keras uses
         # we need to find them since we want to make sure learning phase flags are set to False
-        if learning_phase_flags is None:
-            self.learning_phase_ops = []
-            for op in self.graph.get_operations():
-                if 'learning_phase' in op.name and op.type == "Const" and len(op.outputs[0].shape) == 0:
-                    if op.outputs[0].dtype == tf.bool:
-                        self.learning_phase_ops.append(op)
-            self.learning_phase_flags = [op.outputs[0] for op in self.learning_phase_ops]
+        if self.graph is not None:
+            if learning_phase_flags is None:
+                self.learning_phase_ops = []
+                for op in self.graph.get_operations():
+                    if 'learning_phase' in op.name and op.type == "Const" and len(op.outputs[0].shape) == 0:
+                        if op.outputs[0].dtype == tf.bool:
+                            self.learning_phase_ops.append(op)
+                self.learning_phase_flags = [op.outputs[0] for op in self.learning_phase_ops]
+            else:
+                self.learning_phase_ops = [t.op for t in learning_phase_flags]
         else:
-            self.learning_phase_ops = [t.op for t in learning_phase_flags]
+            self.learning_phase_ops = self.learning_phase_flags = []
```

Comment:
This is a "dirty" workaround for the fact that later TF versions do not expose ops or a graph anymore

Reply:
The workaround here is just to not look for learning phase flags....this could cause problems if we need to find those flags during the eval process.